protocol/bc/issuance_witness.go

```diff
@@ -1,9 +1,78 @@
 package bc
 
+import (
+	"fmt"
+	"io"
+
+	"chain/crypto/sha3pool"
+	"chain/encoding/blockchain"
+	"chain/errors"
+)
+
 type IssuanceWitness struct {
 	InitialBlock    Hash
 	AssetDefinition []byte
 	VMVersion       uint64
 	IssuanceProgram []byte
 	Arguments       [][]byte
 }
+
+func (iw *IssuanceWitness) AssetID() AssetID {
+	return ComputeAssetID(iw.IssuanceProgram, iw.InitialBlock, iw.VMVersion, iw.AssetDefinitionHash())
+}
+
+func (iw *IssuanceWitness) AssetDefinitionHash() (defhash Hash) {
+	if len(iw.AssetDefinition) == 0 {
```

Comment:
No, this is just an optimization. The test vectors are changing because of the fixes to https://github.com/chain/chain/issues/330 and https://github.com/chain/chain/issues/335.